src/crypto.js

```diff
@@ -0,0 +1,278 @@
+  * @param {string} [mode] - The mode of the key to import (default 'AES-GCM')
+  * @returns {arrayBuffer|Object} - The cryptoKey
+  */
+const importKey = (key, type = 'raw', mode = 'AES-GCM') => {
+  return window.crypto.subtle.importKey(type, key, { name: mode }
+    , true, ['encrypt', 'decrypt'])
+}
+
+/**
+ * Decrypt buffer
+ *
+ * @param {ArrayBuffer} data - Data to decrypt
+ * @param {ArrayBuffer} key - The AES key as raw data. 128 or 256 bits
+ * @param {Object} cipherContext - The AES cipher parameters
+ * @param {ArrayBuffer} cipherContext.iv - The IV
+ * @param {string} cipherContext.name - The encryption mode
+ * @param {ArrayBuffer} [cipherContext.additionalData] - The non-secret authenticated data (only AES-GCM)
+ * @param {ArrayBuffer} [cipherContext.counter] - The counter used for aes-ctr mode
+ * @returns {ArrayBuffer} - The decrypted buffer
+ */
+const decryptBuffer = (data, key, cipherContext) => {
+  // TODO: test input params
+  return window.crypto.subtle.decrypt(cipherContext, key, data)
+    .then(result => new Uint8Array(result))
+}
+
+/**
+ * Encrypt buffer
+ *
+ * @param {ArrayBuffer} data - Data to encrypt
+ * @param {ArrayBuffer} key - The AES CryptoKey
+ * @param {Object} cipherContext - The AES cipher parameters
+ * @param {ArrayBuffer} cipherContext.iv - The IV
+ * @param {string} cipherContext.name - The encryption mode
+ * @param {ArrayBuffer} [cipherContext.additionalData] - The non-secret authenticated data (only AES-GCM)
+ * @param {ArrayBuffer} [cipherContext.counter] - The counter used for aes-ctr mode
+ * @returns {ArrayBuffer} - The encrypted buffer
+ */
+const encryptBuffer = (data, key, cipherContext) => {
+  return window.crypto.subtle.encrypt(cipherContext, key, data)
+    .then(result => new Uint8Array(result))
+}
+
+/**
+ * Encrypt data
+ *
+ * @param {CryptoKey} key - The AES CryptoKey
+ * @param {string | Object} - The data to encrypt
+ * @param {string} [format] - The ciphertext and iv encoding format
+ * @returns {Object} - The stringified ciphertext object (ciphertext and iv)
+ */
+const encrypt = (key, data, format = 'hex') => {
+  _checkCryptokey(key)
+  let context = {}
+  let cipherContext = {}
+  context.iv = genRandomBuffer(16)
+  context.plaintext = Buffer.from(JSON.stringify(data))
+
+  // Prepare cipher context, depends on cipher mode
+  cipherContext.name = key.algorithm.name
+  cipherContext.iv = context.iv
+  return encryptBuffer(context.plaintext, key, cipherContext)
+    .then(result => {
+      return {
+        ciphertext: Buffer.from(result).toString(format),
+        iv: Buffer.from(context.iv).toString(format)
+      }
+    })
+}
+
+/**
+ * Decrypt data
+ *
+ * @param {CryptoKey} key - The AES CryptoKey
+ * @param {string | Object} - The data to encrypt
+ * @param {string} [format] - The ciphertext and iv encoding format
+ */
+const decrypt = (key, ciphertext, format = 'hex') => {
+  _checkCryptokey(key)
+  let context = {}
```

Comment:
can you define the object properties at the same time the object is created, like this: `const context = { ciphertext: ..., iv:  ...}

Reply:
ok done in last commit